src/SkyDrop.Core/ViewModels/DropViewModel.cs

```diff
@@ -135,23 +140,55 @@ private async Task StartSendFile()
             IsSendButtonGreen = true;
             IsReceiveButtonGreen = false;
 
-            var file = "Select File";
+            var file = "Select Files";
             var image = "Select Image";
+            var video = "Select Video";
             var cancel = "cancel";
-            var fileType = await userDialogs.ActionSheetAsync("", cancel, "", null, file, image);
+            var fileType = await userDialogs.ActionSheetAsync("", cancel, "", null, file, image, video);
             if (fileType == cancel)
             {
                 ResetUI();
                 return;
             }
-
-            if (fileType == file)
-            {
-                await SelectFileAsyncFunc();
-            }
             else
             {
-                await SelectImageAsyncFunc();
+                SkyFilePickerType chosenType;
+                if (fileType == image)
+                    chosenType = SkyFilePickerType.Image;
+                else if (fileType == video)
+                    chosenType = SkyFilePickerType.Video;
+                else
+                    chosenType = SkyFilePickerType.Generic;
+
+                var pickedFiles = await fileSystemService.PickFilesAsync(chosenType);
+
+                if (pickedFiles == null || pickedFiles.Count() == 0)
+                {
+                    Log.Trace("No file was picked.");
+                    ResetUI();
+
+                    return;
+                }
+
+                var firstFile = pickedFiles.First();
+
+                byte[] fileBytes = null;
+                using (var stream = await firstFile.OpenReadAsync())
```

Comment:
NullReferenceException thrown here when user presses hardware back button while browsing for a file